src/com/google/cose/Ec2SigningKey.java

```diff
@@ -118,69 +118,41 @@ public ECPublicKey getPublicKey() {
     return (ECPublicKey) this.keyPair.getPublic();
   }
 
-  // Big endian: Do not reuse for little endian encodings
-  private static byte[] arrayFromBigNum(BigInteger num, int keySize)
-      throws IllegalArgumentException {
-    // Roundup arithmetic from bits to bytes.
-    byte[] keyBytes = new byte[(keySize + 7) / 8];
-    byte[] keyBytes2 = num.toByteArray();
-    if (keyBytes.length == keyBytes2.length) {
-      return keyBytes2;
-    }
-    if (keyBytes2.length > keyBytes.length) {
-      // There should be no more than one padding(0) byte, invalid key otherwise.
-      if (keyBytes2.length - keyBytes.length > 1 && keyBytes2[0] != 0) {
-        throw new IllegalArgumentException();
-      }
-      System.arraycopy(keyBytes2, keyBytes2.length - keyBytes.length, keyBytes, 0, keyBytes.length);
-    } else {
-      System.arraycopy(
-          keyBytes2, 0, keyBytes, keyBytes.length - keyBytes2.length, keyBytes2.length);
-    }
-    return keyBytes;
-  }
-
   /**
    * Generates a COSE formatted Ec2 signing key given a specific algorithm. The selected key size is
    * chosen based on section 6.2.1 of RFC 5656
    */
   public static Ec2SigningKey generateKey(Algorithm algorithm) throws CborException, CoseException {
-    KeyPair keyPair;
-    int keySize;
     int header;
-    String curveName;
-
+    CurveType curveType;
     switch (algorithm) {
       case SIGNING_ALGORITHM_ECDSA_SHA_256:
-        curveName = "secp256r1";
-        keySize = 256;
+        curveType = CurveType.NIST_P256;
         header = Headers.CURVE_EC2_P256;
         break;
 
       case SIGNING_ALGORITHM_ECDSA_SHA_384:
-        curveName = "secp384r1";
-        keySize = 384;
+        curveType = CurveType.NIST_P384;
         header = Headers.CURVE_EC2_P384;
         break;
 
       case SIGNING_ALGORITHM_ECDSA_SHA_512:
-        curveName = "secp521r1";
-        keySize = 521;
+        curveType = CurveType.NIST_P521;
         header = Headers.CURVE_EC2_P521;
         break;
 
       default:
         throw new CoseException("Unsupported algorithm curve: " + algorithm.getJavaAlgorithmId());
     }
+
+    KeyPair keyPair;
     try {
-      ECGenParameterSpec paramSpec = new ECGenParameterSpec(curveName);
-      KeyPairGenerator gen = KeyPairGenerator.getInstance("EC");
-      gen.initialize(paramSpec);
-      keyPair = gen.genKeyPair();
+      keyPair = generateKeyPair(curveType);
 
       ECPoint pubPoint = ((ECPublicKey) keyPair.getPublic()).getW();
-      byte[] x = arrayFromBigNum(pubPoint.getAffineX(), keySize);
-      byte[] y = arrayFromBigNum(pubPoint.getAffineY(), keySize);
+      int keySize = fieldSizeInBytes(getCurveSpec(curveType).getCurve());
```

Comment:
nit: Not sure if we would benefit from computing the value vs having a fixed value here.

Reply:
Let me know your preference. The current CL preferred fewer magic values.